core/component.go

```diff
@@ -494,3 +536,46 @@ func (c *Component) GetAccessTokens() (tokens map[string]string, err error) {
 	}
 	return tokens, err
 }
+
+// GetStaticComponentPath returns the static path if a component is of type 'static', if not, it returns an empty string
+func (c *Component) GetStaticComponentPath(startPath string)(componentPath string){
+	if c.ComponentType != "static" {
+		return ""
+	}
+
+	if IsValidRemoteComponentConfig(*c) {
+		return path.Join(startPath, "components", c.Name)
+	}
+
+	return path.Join(c.PhysicalPath, c.Path)
+}
+
+
+// CreateDirectory a directory in the given path and reports no error if the directory already exists
+func CreateDirectory(cmdDir string, dirPath string) (componentPath string, err error) {
+	
+	cmd := exec.Command("sh", "-c", "mkdir -p " + dirPath)
+	cmd.Dir = cmdDir
+
+	output, err := cmd.CombinedOutput()
+
+	if err != nil {
+		logger.Error(emoji.Sprintf(":no_entry_sign: Error occurred in creating directory for component\n"))
+		return "", err
+	}
+	if len(output) > 0 {
+		outstring := emoji.Sprintf(":mag_right: Completed creating directory for component\n")
+		logger.Trace(strings.TrimSpace(outstring))
+	}
+
+	return path.Join(cmdDir, dirPath), nil
+}
+
+// IsValidRemoteComponentConfig checks if the given component configuration is valid for a remote component
+func IsValidRemoteComponentConfig(c Component) (bool){
+	return (
+		(c.ComponentType == "static" &&
+		 c.Method == "remote-url")) &&
```

Comment:
I'm starting to think we should call `remote-url` something else. Seeing as `method: git` is technically a _remote url_, we should choose something more concise to what is actually being used. Maybe something like `method: http`? thoughts?